realm/realm-library/src/main/java/io/realm/RealmDictionary.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.realm.internal.OsMap;
+
+/**
+ * Specialization for {@link RealmMap}s whose keys are strings.
```

Comment:
We probably need a longer description for this describing any restrictions and performance considerations. See e.g. RealmList Javadoc